### ISSUE TITLE
Non-existent cwd doesn't error

### ISSUE THREAD

**Issue by dadamssg** [steelbrain/node-ssh#181]
If you specify a non-existent directory via `{cwd: '/foobar'}`, the error is swallowed and the cwd ends up being `/`.

Seems like an easy way for people to shoot themselves in the foot.

**Comment by smeijer** [comment]
Thanks, this one definitely had me fooled. I didn't understand why `{ cwd: '~/subfolder' }` didn't work. Turned out, I needed to change it to `{ cwd: '/root/subfolder' }`.

**Comment by steelbrain** [comment]
Happy to accept a PR for this :)

**Comment by dadamssg** [comment]
@steelbrain it seems you're intentionally hiding the error here : https://github.com/steelbrain/node-ssh/blob/master/src/index.js#L146-L147

I could submit a pr to remove the `1> /dev/null 2> /dev/null` bit but i'm hesitant to believe it would be that simple...

**Comment by steelbrain** [comment]
@dadamssg The reason it works that way is because shell output is very hard to distinguish with. We don't know if an error comes in through an app's output or through `cd`, so we could treat an output from a program as a failure to `cd`. We could ammend that check to see if a file exists before doign a cd, and if it does, exiting with a specific error code and maybe checking for that instead

It's not pretty, but it may work

**Comment by dadamssg** [comment]
i would argue that you don't _need_ to distinguish where the error comes from. I see the `cwd` option as just a prettier way of inlining that into your command.

```sh
ssh.execCommand('hh_client --json', { cwd:'/var/www' })

# should behave the same as

ssh.execCommand('cd /var/www; hh_client --json')
```

**Comment by steelbrain** [comment]
That sounds reasonable to me, tho it would have to be a semver-major change since it'll break existing expectations

**Comment by steelbrain** [comment]
Published in v8.0.0
